src/content.js

```diff
@@ -240,13 +355,38 @@ Function used to avoid multiple injection (cleaner than using an if?)
         }
         OCR = true;
         info('enabling OCR');
+        
+        // Process existing images
         document.querySelectorAll('img').forEach((img) => {
             processImage(img);
         })
         document.querySelectorAll('canvas').forEach((canvas) => {
             processImage(canvas);
         })
-        observer.observe(document.body, {childList: true, subtree: true});
+        
+        // Start observing with more comprehensive options
+        observer.observe(document.body, {
+            childList: true, 
+            subtree: true,
+            attributes: true,
+            attributeFilter: ['src', 'data-src', 'class']
+        });
+        
+        // Also set up a periodic check for dynamically loaded images
+        // This is a fallback for cases where MutationObserver might miss something
+        const periodicCheck = setInterval(() => {
+            if (!OCR) {
+                clearInterval(periodicCheck);
+                return;
+            }
+            
+            document.querySelectorAll('img, canvas').forEach((img) => {
+                if (!processedImages.has(img)) {
```

Comment:
```suggestion
                if (!processedImages.has(img) && !processingImages.has(img)) {
```

We can either do both checks in order to print the message `found unprocessed image during periodic check` or avoid both and defer to the same checks being done inside the `processImage` function